### PR TITLE
Dtsbp 83

### PR DESCRIPTION
**Pull request by rishikrsharma**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBP-83

### Change description ###
Returning number of copies requested

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

**Review by SomeshRA — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 1]
```diff
@@ -178,6 +178,21 @@ private PGPPublicKey loadPgpPublicKey(String encryptionPublicKey) {
...
+        } else if (letter instanceof LetterWithPdfsRequest) {
+            return ((LetterWithPdfsRequest) letter).documents.size();
+        } else if (letter instanceof LetterWithPdfsAndNumberOfCopiesRequest) {
+            return copies.apply((LetterWithPdfsAndNumberOfCopiesRequest) letter);
+        } else {
+            throw new UnsupportedLetterRequestTypeException();
+        }
+    }
+
+    private Function<LetterWithPdfsAndNumberOfCopiesRequest, Integer> copies =
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java:193]: Use - ToIntFunction<LetterWithPdfsAndNumberOfCopiesRequest>

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 2]
```diff
@@ -178,6 +178,21 @@ private PGPPublicKey loadPgpPublicKey(String encryptionPublicKey) {
         }
     }
 
+    private int getCopies(ILetterRequest letter) {
+        if (letter instanceof LetterRequest) {
+            return ((LetterRequest) letter).documents.size();
+        } else if (letter instanceof LetterWithPdfsRequest) {
+            return ((LetterWithPdfsRequest) letter).documents.size();
+        } else if (letter instanceof LetterWithPdfsAndNumberOfCopiesRequest) {
+            return copies.apply((LetterWithPdfsAndNumberOfCopiesRequest) letter);
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java:187]: Only one condition covered

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 3]
```diff
@@ -178,6 +178,21 @@ private PGPPublicKey loadPgpPublicKey(String encryptionPublicKey) {
...
+        } else {
+            throw new UnsupportedLetterRequestTypeException();
```
Comment (SomeshRA) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java:189]: This doesnt have code coverage

**Review by SomeshRA — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java (hunk 1 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: Function applied.

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java (hunk 2 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: Test added.

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java (hunk 3 above)
Comment (rishikrsharma) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: as exception will not come, will break in previous call itself. So removed this line.

**Review by SomeshRA — APPROVED** [review]

**Pull request merged**